src/main/java/com/redhat/red/build/finder/BuildFinder.java

```diff
@@ -380,20 +380,23 @@ private KojiBuild findBestBuild(List<KojiBuild> foundBuilds, List<KojiArchiveInf
         }
 
         if (archiveExtensions == null) {
+            LOGGER.info("Asking Koji for valid archive extensions");
             archiveExtensions = getArchiveExtensions();
         }
 
-        final String EMPTY_MD5 = Hex.encodeHexString(DigestUtils.getDigest(config.getChecksumType().getAlgorithm()).digest());
+        LOGGER.info("Using archive extensions: {}", green(archiveExtensions));
+
+        final String EMPTY_DIGEST = Hex.encodeHexString(DigestUtils.getDigest(config.getChecksumType().getAlgorithm()).digest());
 
         for (Entry<String, Collection<String>> entry : checksumTable.entrySet()) {
               String checksum = entry.getKey();
 
-            if (checksum.equals(EMPTY_MD5)) {
+            if (checksum.equals(EMPTY_DIGEST)) {
                 LOGGER.debug("Found empty file for checksum {}", checksum);
                 continue;
             }
 
-            Collection<String> filenames = checksumTable.get(checksum);
+            Collection<String> filenames = entry.getValue();
```

Comment:
This should be using the existing entry and not getting the value from the map a second time.